assets/src/models/createDetourMachine.ts

```diff
@@ -1,17 +1,142 @@
-import { setup } from "xstate"
+import { setup, assign, fromPromise } from "xstate"
+import { RouteId, RoutePattern, RoutePatternId } from "../schedule"
+import { fetchRoutePatterns } from "../api"
 
 export const createDetourMachine = setup({
   types: {} as {
-    events: { type: "detour.edit.done" } | { type: "detour.edit.resume" }
+    context: {
+      routeId?: RouteId
+      routePatternId?: RoutePatternId
+
+      routePatterns?: RoutePattern[]
+    }
+
+    input:
+      | {
+          // Caller has target route pattern
+          routeId: RouteId
+          routePatternId: RoutePatternId
+        }
+      | {
+          // Caller has target route
+          routeId: RouteId
+          routePatternId: undefined
+        }
+      | {
+          // Caller has no prior selection
+          routeId: undefined
+          routePatternId: undefined
+        }
+
+    events:
+      | { type: "detour.edit.done" }
+      | { type: "detour.edit.resume" }
+      | { type: "detour.route-pattern.open" }
+      | { type: "detour.route-pattern.done" }
+      | { type: "detour.route-pattern.select-route"; routeId: RouteId }
+      | {
+          type: "detour.route-pattern.select-pattern"
+          routePatternId: RoutePatternId
+        }
+  },
+  actors: {
+    "fetch-route-patterns": fromPromise<
+      Awaited<ReturnType<typeof fetchRoutePatterns>>,
+      { routeId?: RouteId }
+    >(async ({ input: { routeId } }) => {
+      if (routeId) {
+        return fetchRoutePatterns(routeId)
+      } else {
+        // Hmm, this could leave us in the lurch if we don't handle it,
+        // I'd like to make this impossible but I can't get typescript to infer
+        // the context types but it's not letting me
```

Comment:
note: I _can_ make the `input` to this actor typed so that you can't call it without a `routeId`, but xstate currently doesn't support type deduction via guards, which is the only way of doing conditional calls, so we end up needing to use the `!` operator, which our `eslint` config disallows.